matrix_content_scanner/scanner/scanner.py

```diff
@@ -111,6 +212,34 @@ async def scan_file(
 
         return media
 
+    def _get_cache_key_for_file(
+        self,
+        media_path: str,
+        metadata: Optional[JsonDict],
+        thumbnail_params: Optional[Dict[str, List[str]]],
+    ) -> str:
+        """Generates the key to use to store the result for the given media in the result
+        cache.
+
+        The key is computed using the media's `server_name/media_id` path, but also the
+        metadata dict (stringified), in case e.g. the decryption key changes, as well as
+        the parameters used to generate the thumbnail if any (stringified), to
+        differentiate thumbnails from full-sized media.
+        The resulting key is a sha256 hash of the concatenation of these two values.
+
+        Args:
+            media_path: The `server_name/media_id` path of the file to scan.
+            metadata: The file's metadata (or None if the file isn't encrypted).
+            thumbnail_params: The parameters to generate thumbnail with. If no parameter
+                is passed, this will be an empty dict. If the media being requested is not
+                a thumbnail, this will be None.
+        """
+        raw_metadata = json.dumps(metadata)
+        raw_params = json.dumps(thumbnail_params)
```

Comment:
obviously this means that the cache key depends on the order of the thumbnail params... not sure how much of a problem that is in this case.